src/main/java/com/ziminer/xork/world/World.java

```diff
@@ -41,8 +45,50 @@ public static String coordToKey(int x, int y) {
      * @return The newly created {@link World}.
      */
     public static World newGridWorld(int width, int height) {
-        // TODO (roman, Sept 14 2016): Make this work.
-        throw new IllegalStateException("NOT WORKING YET.");
+        HashMap<String, Tile> tiles = new HashMap<>();
+        List<Tile> bfs = new ArrayList<>();
+        Tile curr, temp;
+        String newkey;
+        String name = "Grid:(" + width + "," + height + ")";
+        width--; height--;
+
+        if(width >= 0 && height >= 0){
+            Tile first = new Tile(width + "," + height);
+            tiles.put(width + "," + height, first);
+            bfs.add(first);
+        }
+
+        while(!bfs.isEmpty()){
+            curr = bfs.remove(0);
+            width = Integer.parseInt(curr.key.split(",")[0]);
+            height = Integer.parseInt(curr.key.split(",")[1]);
+
+            if(width-1 >= 0){
+                newkey = (width-1) + "," +height;
+                temp = tiles.remove(newkey);
+                if(temp == null) {
+                    temp = new Tile(newkey);
+                    bfs.add(temp);
+                }
+                curr.addTransition(new TileTransition(temp, newkey));
+                temp.addTransition(new TileTransition(curr, curr.key));
+                tiles.put(newkey, temp);
+            }
+
+            if(height-1 >= 0){
```

Comment:
The code in this block and the width-1 block is pretty much the same. Maybe put it in a helper function?

Reply:
Was thinking about it, will put it in